Documentation: include E2E documentation in devdocs.

worldomonation opened the pull request:
#### Changes proposed in this Pull Request

This PR adds a new section in the devdocs named `Testing` and incorporates links to the E2E documentation in the devdocs.

Key changes:
- improved formatting of the sidebar.
- incorporate links to E2E documentation from the devdocs.

#### Testing instructions

Navigate to https://container-determined-bhabha.calypso.live/devdocs and check out the Testing category.


Closes https://github.com/Automattic/wp-calypso/issues/61182.

worldomonation committed:
Update devdocs:
- add new section `Testing`
- reorganize existing `docs/testing` a bit to make more sense.
- link to `test/e2e/README.md` from the Testing category.
- use cloud hosted images instead

worldomonation committed:
Remove images

dpasque commented on docs/testing/end-to-end-tests.md:
```diff
@@ -0,0 +1,31 @@
+# Overview
+
+<!-- TOC -->
+
+- [Overview](#overview)
+    - [What is this?](#what-is-this)
+    - [Our Goals](#our-goals)
+    - [What is tested?](#what-is-tested)
+
+<!-- /TOC -->
+
+## What is this?
+
+End-to-end tests (`e2e` for short) describes the automated functional tests that simulate user interaction with the system under test.
+
+A well-maintained suite of e2e tests, in conjunction with CI (continuous integration) pipelines will ensure that regressions to key user flows are caught as early as possible in the development cycle.
+
+## Our Goals
+
+To accelerate development by being a force for continuous improvement, and help identify, prioritize and mitigate bottlenecks from the system.
+
+<sup>(taken from Quality Squad internal memo)</sup>
+
+## What is tested?
+
+These end-to-end test focus on several key areas:
+
+- Calypso features.
+- interaction between Calypso and Gutenberg editor.
+- internationalization and localization.
+- Gutenberg blocks added by WordPress.com.
```
Out of curiosity, what's the purpose of this specific doc? It doesn't seem like it's linked to from anywhere. Could we just add any of these bits to the overview doc in `/test/e2e`?

worldomonation replied:
Hmm, I don't recall anymore, it may be an artifact from when I previously looked at moving the entire `test/e2e` directory.

worldomonation replied:
Removed.

dpasque commented on docs/testing/index.md:
```diff
@@ -0,0 +1,16 @@
+# Testing Guide
```
This seems like the new entry point for testing -- could we replace the link [here](https://github.com/Automattic/wp-calypso/blob/trunk/docs/guide/index.md?plain=1#L11) to point to this page?

worldomonation replied:
Fixed - replaced the link in the Calypso Guide to point to `testing/index.md`.

dpasque commented on docs/testing/testing-overview.md:
````diff
@@ -73,56 +61,8 @@ They run daily on continuous integration (TeamCity), because they can use networ
-Behind the scenes we maintain 3 test configuration. This is because each of them (`client`, `server`, and `integration`) has their own requirements.
+A well-maintained suite of e2e tests, in conjunction with CI (continuous integration) pipelines will ensure that regressions to key user flows are caught as early as possible in the development cycle.
 
-### How to run a smaller subset of test files?
-
-We have a yarn run script for each tests type: `yarn run test-client`, `yarn run test-server`, `yarn run test-integration`.
-You can pass a filename, folder name or matching pattern to these scripts to narrow down number of executed tests.
-
-Example for client:
-
-```bash
-> # run test suite for a specific test file
-> yarn run test-client client/state/selectors/test/get-media.js
-> # run test suites for all files in a specific folder
-> yarn run test-server server/config
-> # run test suites for all files matching pattern
-> yarn run test-client client/*/domains
-```
-
-### How to run specified suite or test-case
-
-The exclusivity feature allows you to run only the specified suite or test-case by appending `.only()` to the function.
-It works with `describe` and `it` functions. More details in [Jest documentation](https://facebook.github.io/jest/docs/api.html).
-
-Using `only` is a little bit dangerous, as you may end up committing the `only`, which would cause the test suite to only run your test on the build server. So be sure to look for stray only calls when reviewing a test. We have [eslint rules](https://github.com/jest-community/eslint-plugin-jest) that should catch them for you.
-
-Example:
-
-```js
-describe.only( 'just run this suite', function () {
-	test( 'should run these tests', function () {
-		// your test
-	} );
-
-	test.only( 'should only run this one test', function () {
-		// just run this test if the only is present
-	} );
-} );
-```
````
Could you explain a little more about why we should pull this content out? Do these commands not work anymore?

worldomonation replied:
I think what you're seeing is artifact of the in-between state the repo was in, when I tried to combine `test/e2e` into this directory.

worldomonation replied:
Looking at the context a bit more, it looks like my intention was to make the overview page more generic and move the specifics like this above into its own pages.

worldomonation replied:
Moved this whole FAQ section under its own article, `faq.md`.

dpasque commented on client/devdocs/sidebar.jsx:
```diff
@@ -87,7 +87,15 @@ export default class DevdocsSidebar extends PureComponent {
+						selected={ this.isItemSelected( '/devdocs/docs/testing', false ) }
+					/>
 
+					<br></br>
```
I believe we're supposed to avoid using `br` for style and layout purposes. What do you think instead of adding a class to the headings that need extra space above them and adding `margin` in the [devdocs styles](https://github.com/Automattic/wp-calypso/blob/trunk/client/devdocs/style.scss)?

worldomonation replied:
I fixed this by using `SidebarSeparator` class and then adding a 20px padding on the class in devdocs stylesheet.

dpasque reviewed (commented):
Overall I really like the changes! Just have a few questions for clarification! Thanks for doing this! :)

worldomonation committed:
- add spacing in the sidebar using `sidebar__separator`
- add page for FAQ
- fix links to the guides

worldomonation committed:
- run prettier
- re-add missing section in style guide about having only one top-level describe block.

dpasque reviewed (approved):
Looks great!! Thanks for the updates! 🚀
